packages/multicast_dns/lib/src/lookup_resolver.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015, the Dartino project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:multicast_dns/src/resource_record.dart';
+
+/// Class for maintaining state about pending mDNS requests.
+class PendingRequest extends LinkedListEntry<PendingRequest> {
+  /// Creates a new PendingRequest.
+  PendingRequest(this.type, this.name, this.controller);
+
+  /// The [ResourceRecordType] of the request.
+  final int type;
+
+  /// The domain name.
+  final String name;
+
+  /// A StreamController managing the request.
+  final StreamController<ResourceRecord> controller;
+
+  /// The timer for the request.
+  Timer timer;
+}
+
+/// Class for keeping track of pending lookups and process incoming
+/// query responses.
+class LookupResolver {
+  /// The requests the process.
```

Comment:
Huh? This sentence makes no sense.  Maybe "requests the process." -> "pending requests to be processed."? Also, doesn't add a lot of information. What is processing the requests, etc.